docs/declaring-loaders.rst

```diff
@@ -5,22 +5,40 @@ Declaring Item Loaders
 
 Item Loaders are declared by using a class definition syntax. Here is an example::
 
+    from dataclasses import dataclass, field
+
     from itemloaders import ItemLoader
-    from itemloaders.processors import TakeFirst, MapCompose, Join
+    from itemloaders.processors import TakeFirst
 
-    class ProductLoader(ItemLoader):
 
-        default_output_processor = TakeFirst()
+    def price_out(values):
+        return float(values[0])
+
+
+    @dataclass(init=False)
+    class Product:
+        name: str
+        price: float = field(
+            metadata={
+            'output_processor': price_out
+            }
+        )
```

Comment:
I wonder if that's possible to sidestep the issues in this example by using dict as an item.

Reply:
Not sure about the `dict`, as one of the ideas of this example is to show the metadata processors.
Maybe using `field(default=None)` then

Reply:
The text below this example is not discussing output_processor defined in field meta, so it could be fine.

Reply:
Not the one exactly above, but 5 lines below, stating the precedence.
And it would be the only place where this feature is shown

Reply:
Hey! I start to re-read docs again, and this piece still bothers me :( This is a documentation part which introduces item loader and shows the very basics - how to declare it, and yet it uses dataclasses, which are not considered experimental, and which have caveats with default fields.

Reply:
Updated!